src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3740,6 +3769,21 @@ void os::large_page_init() {
   // Query OS information first.
   HugePages::initialize();
 
+  // If THPs are unconditionally enabled (THP mode "always"), khugepaged may attempt to
+  // coalesce small pages in thread stacks to huge pages. That costs a lot of memory and
+  // is usually unwanted for thread stacks. Therefore we attempt to prevent THP formation in
+  // thread stacks unless the user explicitly allowed THP formation by manually disabling
+  // -XX:+DisableTHPStackMitigation.
+  if (HugePages::thp_mode() == THPMode::always) {
```

Comment:
What controls this? The user launching the VM? The OS configuration? Or both?

Reply:
The OS, so ultimately the Admin that set up the OS.